server/bound_queries.cc

```diff
@@ -50,13 +50,48 @@ struct BoundSelectQuery::Executor {
                          bool processInParallel,
                          ssize_t offset,
                          ssize_t limit,
-                         std::function<bool (const Json::Value &)> onProgress) = 0;
+                         std::function<bool (const Json::Value &)> onProgress)
+    {
+        auto newProcessor = [&] (Path & rowName,
+                                 ExpressionValue & val,
+                                 std::vector<ExpressionValue> & calcd,
+                                 int bucketNum)
+            {
+                NamedRowValue output;
+                output.rowName = std::move(rowName);
+                output.rowHash = output.rowName;
```

Comment:
It's OK.  There is a conversion operator; it's a supported way of filling in a row hash when only a row name is available.